pretyping/coercion.ml

```diff
@@ -45,35 +45,31 @@ exception NoCoercion
 exception NoCoercionNoUnifier of evar_map * unification_error
 
 (* Here, funj is a coercion therefore already typed in global context *)
-let apply_coercion_args env sigma check isproj argl funj =
-  let rec apply_rec sigma acc typ = function
-    | [] ->
-      (match isproj with
-       | Some p ->
+let apply_coercion_args env sigma isproj bo ty arg arg_ty evars =
```

Comment:
I find `evars` to be about what you do with this rather than what it is is. IMO `nparams` would be better.

Also, this piece of code seems to be simpler if split into 2 steps:
- saturation (apply a term of functional type to n evars standing for its arguments, obtain the resulting type and term)
- all the rest (unify that type with a given one, do something if it is a proj...)

Reply:
Indeed, done